Kytos is sending the incorrect topologia to SDX-LC via push when Interface metadata changes

As we discussed on the PR https://github.com/atlanticwave-sdx/sdx-controller/pull/514, it may happen that Kytos sends the incorrect topology version to SDX-LC.

Here is how to reproduce:

```
$ git clone https://github.com/atlanticwave-sdx/sdx-end-to-end-tests
$ cd sdx-end-to-end-tests
$ git diff
diff --git a/docker-compose.yml b/docker-compose.yml
index 8b430f2..6cdae98 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -26,6 +26,7 @@ services:
       - .env
     volumes:
       - .:/sdx-end-to-end-tests
+      - ./kytos-sdx:/src/kytos-sdx
     depends_on:
       mongo:
         condition: service_healthy
diff --git a/tests/test_21_use_case_topology.py b/tests/test_21_use_case_topology.py
index c806a36..155b41f 100644
--- a/tests/test_21_use_case_topology.py
+++ b/tests/test_21_use_case_topology.py
@@ -306,7 +306,7 @@ class TestE2ETopologyUseCases:
         l2vpn_status = l2vpn_data.get("status")
         assert l2vpn_status == "down", l2vpn_data

-    @pytest.mark.xfail(reason="The link status is up after changing the attribute sdx_nni='' on one side")
+    #@pytest.mark.xfail(reason="The link status is up after changing the attribute sdx_nni='' on one side")
     def test_091_port_missing_nni(self):
         """
         Use Case 9: OXPO sends a topology update with a Port missing
@@ -338,6 +338,14 @@ class TestE2ETopologyUseCases:

         time.sleep(5)

+        response = requests.get(f"{sdx_api}/topology/2.0.0")
+        kytos_topo = response.json()
+        for node in kytos_topo["nodes"]:
+            for port in node["ports"]:
+                if port["id"] == "urn:sdx:port:ampath.net:Ampath1:40":
+                    assert port["nni"] == ""
+                    break
+
         response = requests.get(API_URL_TOPO)
         assert response.status_code == 200, response.text
         updated_topology = response.json()
@@ -345,7 +353,7 @@ class TestE2ETopologyUseCases:
             if node['name'] == port_name.split('-')[0]:
                 for port in node['ports']:
                     if port['name'] == port_name:
-                        assert port['nni'] == ''
+                        assert port['nni'] == '', f"sdx_topo={updated_topology} kytos_topo={kytos_topo}"
         for link in updated_topology['links']:
             if link['name'] == link_name:
                 assert link['status'] == 'error'
$ git clone https://github.com/atlanticwave-sdx/kytos-sdx
$ cd kytos-sdx
$ git diff
diff --git a/main.py b/main.py
index 988d59d..09939f9 100644
--- a/main.py
+++ b/main.py
@@ -414,6 +414,7 @@ class Main(KytosNApp):  # pylint: disable=R0904
     @rest("topology/2.0.0", methods=["POST"])
     def send_topology_to_sdxlc(self, _request: Request) -> JSONResponse:
         """Send the topology (v2) to SDX-LC"""
+        log.info(f"troubleshooting: topology={self._converted_topo}")
         self.post_topology_to_sdxlc(self._converted_topo)
         return JSONResponse("Operation successful", status_code=200)
$ cd ..
$ docker compose down -v; docker compose up -d && ./wait-mininet-ready.sh && docker compose exec -it mininet python3 -m pytest tests/test_21_use_case_topology.py::TestE2ETopologyUseCases::test_091_port_missing_nni
[+] Running 11/10
 ✔ Container sdx-end-to-end-tests-tenet-lc-1        Removed                                                                                                                                                           13.3s
 ✔ Container sdx-end-to-end-tests-sax-lc-1          Removed                                                                                                                                                           12.2s
 ✔ Container sdx-end-to-end-tests-sdx-controller-1  Removed                                                                                                                                                            2.1s
 ✔ Container sdx-end-to-end-tests-mininet-1         Removed                                                                                                                                                           12.2s
 ✔ Container sdx-end-to-end-tests-ampath-lc-1       Removed                                                                                                                                                           13.3s
 ✔ Container sdx-end-to-end-tests-ampath-1          Removed                                                                                                                                                           12.4s
 ✔ Container sdx-end-to-end-tests-sax-1             Removed                                                                                                                                                           12.4s
 ✔ Container sdx-end-to-end-tests-tenet-1           Removed                                                                                                                                                           13.1s
 ✔ Container sdx-end-to-end-tests-mq1-1             Removed                                                                                                                                                            2.7s
 ✔ Container sdx-end-to-end-tests-mongo-1           Removed                                                                                                                                                            1.6s
 ✘ Network sdx-end-to-end-tests_default             Error                                                                                                                                                              0.0s
failed to remove network sdx-end-to-end-tests_default: Error response from daemon: error while removing network: network sdx-end-to-end-tests_default id 58ab63c922f819324ee6cd7ef0bce2fc739b0030c960e4de725167cf3ba56184 has active endpoints
[+] Running 20/20
 ✔ sax-lc Pulled                                                                                                                                                                                                      15.0s
 ✔ ampath Pulled                                                                                                                                                                                                       3.8s
 ✔ tenet Pulled                                                                                                                                                                                                        4.1s
 ✔ mininet Pulled                                                                                                                                                                                                      3.9s
 ✔ tenet-lc Pulled                                                                                                                                                                                                    15.1s
   ✔ ccaf924377f9 Already exists                                                                                                                                                                                       0.0s
   ✔ f4e881ec6b64 Already exists                                                                                                                                                                                       0.0s
   ✔ 552132316e5d Already exists                                                                                                                                                                                       0.0s
   ✔ 0426d084ccf5 Already exists                                                                                                                                                                                       0.0s
   ✔ 5b3b161a20b8 Pull complete                                                                                                                                                                                        1.9s
   ✔ 64a1dc7b0f26 Pull complete                                                                                                                                                                                        0.2s
   ✔ 4f4fb700ef54 Pull complete                                                                                                                                                                                        0.1s
   ✔ 3e2a7a7dc1fe Pull complete                                                                                                                                                                                        0.3s
   ✔ 5cc4b472978f Pull complete                                                                                                                                                                                        0.6s
   ✔ 7633fa64ca28 Pull complete                                                                                                                                                                                        1.3s
 ✔ mq1 Pulled                                                                                                                                                                                                          4.5s
 ✔ sdx-controller Pulled                                                                                                                                                                                               4.5s
 ✔ ampath-lc Pulled                                                                                                                                                                                                   15.0s
 ✔ mongo Pulled                                                                                                                                                                                                        4.4s
 ✔ sax Pulled                                                                                                                                                                                                          4.0s
[+] Running 10/10
 ✔ Container sdx-end-to-end-tests-mongo-1           Healthy                                                                                                                                                            7.1s
 ✔ Container sdx-end-to-end-tests-mq1-1             Healthy                                                                                                                                                            7.0s
 ✔ Container sdx-end-to-end-tests-sax-1             Started                                                                                                                                                            1.8s
 ✔ Container sdx-end-to-end-tests-sax-lc-1          Started                                                                                                                                                            1.7s
 ✔ Container sdx-end-to-end-tests-ampath-lc-1       Started                                                                                                                                                            1.8s
 ✔ Container sdx-end-to-end-tests-sdx-controller-1  Started                                                                                                                                                            1.8s
 ✔ Container sdx-end-to-end-tests-tenet-1           Started                                                                                                                                                            1.8s
 ✔ Container sdx-end-to-end-tests-ampath-1          Started                                                                                                                                                            1.8s
 ✔ Container sdx-end-to-end-tests-tenet-lc-1        Started                                                                                                                                                            1.8s
 ✔ Container sdx-end-to-end-tests-mininet-1         Started                                                                                                                                                            0.9s
Waiting mininet to be ready...............ok
=================================================================================================== test session starts ====================================================================================================
platform linux -- Python 3.11.2, pytest-7.2.1, pluggy-1.0.0+repack
rootdir: /sdx-end-to-end-tests
plugins: unordered-0.7.0
collected 1 item

tests/test_21_use_case_topology.py F                                                                                                                                                                                 [100%]

========================================================================================================= FAILURES =========================================================================================================
____________________________________________________________________________________ TestE2ETopologyUseCases.test_091_port_missing_nni _____________________________________________________________________________________

self = <test_21_use_case_topology.TestE2ETopologyUseCases object at 0x7f19dbb4f650>

    def test_091_port_missing_nni(self):
        """
        Use Case 9: OXPO sends a topology update with a Port missing
        """
        l2vpn_data = self.create_new_l2vpn(vlan='910', node1='Ampath1', node2= 'Sax01')
        l2vpn_id = l2vpn_data['id']
        port_name = 'Ampath1-eth40'
        link_name = 'Ampath1-eth40--Sax01-eth40'
        interfaces_id = 'aa:00:00:00:00:00:00:01:40'

        # Verify the link is up in the topology
        response = requests.get(API_URL_TOPO)
        assert response.status_code == 200, response.text
        topology = response.json()
        for link in topology['links']:
            if link['name'] == link_name:
                assert link['status'] == 'up'
                break

        ampath_api = KYTOS_API % 'ampath'
        api_url_ampath = f'{ampath_api}/topology/v3'
        response = requests.delete(f"{api_url_ampath}/interfaces/{interfaces_id}/metadata/sdx_nni")
        assert response.status_code == 200, response.text

        # Force to send the topology to the SDX-LC
        sdx_api = KYTOS_SDX_API % 'ampath'
        response = requests.post(f"{sdx_api}/topology/2.0.0")
        assert response.status_code == 200, response.text

        time.sleep(5)

        response = requests.get(f"{sdx_api}/topology/2.0.0")
        kytos_topo = response.json()
        for node in kytos_topo["nodes"]:
            for port in node["ports"]:
                if port["id"] == "urn:sdx:port:ampath.net:Ampath1:40":
                    assert port["nni"] == ""
                    break

        response = requests.get(API_URL_TOPO)
        assert response.status_code == 200, response.text
        updated_topology = response.json()
        for node in updated_topology['nodes']:
            if node['name'] == port_name.split('-')[0]:
                for port in node['ports']:
                    if port['name'] == port_name:
>                       assert port['nni'] == '', f"sdx_topo={updated_topology} kytos_topo={kytos_topo}"
E                       AssertionError: sdx_topo={'id': 'urn:sdx:topology:', 'links': [{'availability': 100, 'bandwidth': 10, 'id': 'urn:sdx:link:interdomain:sax.net:Sax01:41:tenet.ac.za:Tenet01:41', 'latency': 0, 'measurement_period': None, 'name': 'Sax01-eth41--Tenet01-eth41', 'packet_loss': 0, 'ports': ['urn:sdx:port:sax.net:Sax01:41', 'urn:sdx:port:tenet.ac.za:Tenet01:41'], 'private_attributes': None, 'residual_bandwidth': 100, 'short_name': None, 'state': 'enabled', 'status': 'up', 'timestamp': None}, {'availability': 100, 'bandwidth': 10, 'id': 'urn:sdx:link:interdomain:sax.net:Sax02:41:tenet.ac.za:Tenet02:41', 'latency': 0, 'measurement_period': None, 'name': 'Sax02-eth41--Tenet02-eth41', 'packet_loss': 0, 'ports': ['urn:sdx:port:sax.net:Sax02:41', 'urn:sdx:port:tenet.ac.za:Tenet02:41'], 'private_attributes': None, 'residual_bandwidth': 100, 'short_name': None, 'state': 'enabled', 'status': 'up', 'timestamp': None}, {'availability': 100, 'bandwidth': 10, 'id': 'urn:sdx:link:interdomain:ampath.net:Ampath1:40:sax.net:Sax01:40', 'latency': 0, 'measurement_period': None, 'name': 'Ampath1-eth40--Sax01-eth40', 'packet_loss': 0, 'ports': ['urn:sdx:port:ampath.net:Ampath1:40', 'urn:sdx:port:sax.net:Sax01:40'], 'private_attributes': None, 'residual_bandwidth': 100.0, 'short_name': None, 'state': 'enabled', 'status': 'up', 'timestamp': None}, {'availability': 100, 'bandwidth': 10, 'id': 'urn:sdx:link:interdomain:ampath.net:Ampath2:40:sax.net:Sax02:40', 'latency': 0, 'measurement_period': None, 'name': 'Ampath2-eth40--Sax02-eth40', 'packet_loss': 0, 'ports': ['urn:sdx:port:ampath.net:Ampath2:40', 'urn:sdx:port:sax.net:Sax02:40'], 'private_attributes': None, 'residual_bandwidth': 100, 'short_name': None, 'state': 'enabled', 'status': 'up', 'timestamp': None}, {'availability': 100, 'bandwidth': 10, 'id': 'urn:sdx:link:sax.net:Sax01/1_Sax02/1', 'latency': 0, 'measurement_period': None, 'name': 'Sax01/1_Sax02/1', 'packet_loss': 0, 'ports': ['urn:sdx:port:sax.net:Sax01:1', 'urn:sdx:port:sax.net:Sax02:1'], 'private_attributes': None, 'residual_bandwidth': 100, 'short_name': None, 'state': 'enabled', 'status': 'up', 'timestamp': None}, {'availability': 100, 'bandwidth': 10, 'id': 'urn:sdx:link:tenet.ac.za:Tenet01/1_Tenet02/1', 'latency': 0, 'measurement_period': None, 'name': 'Tenet01/1_Tenet02/1', 'packet_loss': 0, 'ports': ['urn:sdx:port:tenet.ac.za:Tenet01:1', 'urn:sdx:port:tenet.ac.za:Tenet02:1'], 'private_attributes': None, 'residual_bandwidth': 100, 'short_name': None, 'state': 'enabled', 'status': 'up', 'timestamp': None}, {'availability': 100, 'bandwidth': 10, 'id': 'urn:sdx:link:tenet.ac.za:Tenet01/2_Tenet03/2', 'latency': 0, 'measurement_period': None, 'name': 'Tenet01/2_Tenet03/2', 'packet_loss': 0, 'ports': ['urn:sdx:port:tenet.ac.za:Tenet01:2', 'urn:sdx:port:tenet.ac.za:Tenet03:2'], 'private_attributes': None, 'residual_bandwidth': 100, 'short_name': None, 'state': 'enabled', 'status': 'up', 'timestamp': None}, {'availability': 100, 'bandwidth': 10, 'id': 'urn:sdx:link:ampath.net:Ampath1/1_Ampath2/1', 'latency': 0, 'measurement_period': None, 'name': 'Ampath1/1_Ampath2/1', 'packet_loss': 0, 'ports': ['urn:sdx:port:ampath.net:Ampath1:1', 'urn:sdx:port:ampath.net:Ampath2:1'], 'private_attributes': None, 'residual_bandwidth': 100, 'short_name': None, 'state': 'enabled', 'status': 'up', 'timestamp': None}, {'availability': 100, 'bandwidth': 10, 'id': 'urn:sdx:link:ampath.net:Ampath2/3_Ampath3/3', 'latency': 0, 'measurement_period': None, 'name': 'Ampath2/3_Ampath3/3', 'packet_loss': 0, 'ports': ['urn:sdx:port:ampath.net:Ampath2:3', 'urn:sdx:port:ampath.net:Ampath3:3'], 'private_attributes': None, 'residual_bandwidth': 100, 'short_name': None, 'state': 'enabled', 'status': 'up', 'timestamp': None}, {'availability': 100, 'bandwidth': 10, 'id': 'urn:sdx:link:ampath.net:Ampath1/2_Ampath3/2', 'latency': 0, 'measurement_period': None, 'name': 'Ampath1/2_Ampath3/2', 'packet_loss': 0, 'ports': ['urn:sdx:port:ampath.net:Ampath1:2', 'urn:sdx:port:ampath.net:Ampath3:2'], 'private_attributes': None, 'residual_bandwidth': 100, 'short_name': None, 'state': 'enabled', 'status': 'up', 'timestamp': None}], 'model_version': '2.0.0', 'name': 'SAX-OXP', 'nodes': [{'id': 'urn:sdx:node:sax.net:Sax01', 'location': {'address': 'Fortaleza', 'iso3166_2_lvl4': 'BR-CE', 'latitude': -3.0, 'longitude': -40.0}, 'name': 'Sax01', 'ports': [{'entities': [], 'id': 'urn:sdx:port:sax.net:Sax01:41', 'name': 'Sax01-eth41', 'nni': 'urn:sdx:port:tenet.ac.za:Tenet01:41', 'node': 'urn:sdx:node:sax.net:Sax01', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': ['1-4094']}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}, {'entities': [], 'id': 'urn:sdx:port:sax.net:Sax01:1', 'name': 'Sax01-eth1', 'nni': 'urn:sdx:link:sax.net:Sax01/1_Sax02/1', 'node': 'urn:sdx:node:sax.net:Sax01', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': ['1-4094']}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}, {'entities': [], 'id': 'urn:sdx:port:sax.net:Sax01:40', 'name': 'Sax01-eth40', 'nni': 'urn:sdx:port:ampath.net:Ampath1:40', 'node': 'urn:sdx:node:sax.net:Sax01', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': ['2-4094']}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}, {'entities': [], 'id': 'urn:sdx:port:sax.net:Sax01:50', 'name': 'Sax01-eth50', 'nni': '', 'node': 'urn:sdx:node:sax.net:Sax01', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': ['1-909', '911-4094']}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}], 'private_attributes': [], 'short_name': None, 'state': 'enabled', 'status': 'up'}, {'id': 'urn:sdx:node:sax.net:Sax02', 'location': {'address': 'Fortaleza', 'iso3166_2_lvl4': 'BR-CE', 'latitude': -3.0, 'longitude': -20.0}, 'name': 'Sax02', 'ports': [{'entities': [], 'id': 'urn:sdx:port:sax.net:Sax02:41', 'name': 'Sax02-eth41', 'nni': 'urn:sdx:port:tenet.ac.za:Tenet02:41', 'node': 'urn:sdx:node:sax.net:Sax02', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': ['1-4094']}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}, {'entities': [], 'id': 'urn:sdx:port:sax.net:Sax02:1', 'name': 'Sax02-eth1', 'nni': 'urn:sdx:link:sax.net:Sax01/1_Sax02/1', 'node': 'urn:sdx:node:sax.net:Sax02', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': ['1-4094']}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}, {'entities': [], 'id': 'urn:sdx:port:sax.net:Sax02:40', 'name': 'Sax02-eth40', 'nni': 'urn:sdx:port:ampath.net:Ampath2:40', 'node': 'urn:sdx:node:sax.net:Sax02', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': ['1-4094']}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}, {'entities': [], 'id': 'urn:sdx:port:sax.net:Sax02:50', 'name': 'Sax02-eth50', 'nni': '', 'node': 'urn:sdx:node:sax.net:Sax02', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': ['1-4094']}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}], 'private_attributes': [], 'short_name': None, 'state': 'enabled', 'status': 'up'}, {'id': 'urn:sdx:node:tenet.ac.za:Tenet01', 'location': {'address': 'CapeTown', 'iso3166_2_lvl4': 'ZA-WC', 'latitude': -33.0, 'longitude': 18.0}, 'name': 'Tenet01', 'ports': [{'entities': [], 'id': 'urn:sdx:port:tenet.ac.za:Tenet01:41', 'name': 'Tenet01-eth41', 'nni': 'urn:sdx:port:sax.net:Sax01:41', 'node': 'urn:sdx:node:tenet.ac.za:Tenet01', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': ['1-4094']}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}, {'entities': [], 'id': 'urn:sdx:port:tenet.ac.za:Tenet01:1', 'name': 'Tenet01-eth1', 'nni': 'urn:sdx:link:tenet.ac.za:Tenet01/1_Tenet02/1', 'node': 'urn:sdx:node:tenet.ac.za:Tenet01', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': ['1-4094']}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}, {'entities': [], 'id': 'urn:sdx:port:tenet.ac.za:Tenet01:2', 'name': 'Tenet01-eth2', 'nni': 'urn:sdx:link:tenet.ac.za:Tenet01/2_Tenet03/2', 'node': 'urn:sdx:node:tenet.ac.za:Tenet01', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': ['1-4094']}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}, {'entities': [], 'id': 'urn:sdx:port:tenet.ac.za:Tenet01:50', 'name': 'Tenet01-eth50', 'nni': '', 'node': 'urn:sdx:node:tenet.ac.za:Tenet01', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': ['1-4094']}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}], 'private_attributes': [], 'short_name': None, 'state': 'enabled', 'status': 'up'}, {'id': 'urn:sdx:node:tenet.ac.za:Tenet02', 'location': {'address': 'Johanesburgo', 'iso3166_2_lvl4': 'ZA-GP', 'latitude': -26.0, 'longitude': 28.0}, 'name': 'Tenet02', 'ports': [{'entities': [], 'id': 'urn:sdx:port:tenet.ac.za:Tenet02:41', 'name': 'Tenet02-eth41', 'nni': 'urn:sdx:port:sax.net:Sax02:41', 'node': 'urn:sdx:node:tenet.ac.za:Tenet02', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': ['1-4094']}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}, {'entities': [], 'id': 'urn:sdx:port:tenet.ac.za:Tenet02:1', 'name': 'Tenet02-eth1', 'nni': 'urn:sdx:link:tenet.ac.za:Tenet01/1_Tenet02/1', 'node': 'urn:sdx:node:tenet.ac.za:Tenet02', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': ['1-4094']}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}, {'entities': [], 'id': 'urn:sdx:port:tenet.ac.za:Tenet02:50', 'name': 'Tenet02-eth50', 'nni': '', 'node': 'urn:sdx:node:tenet.ac.za:Tenet02', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': ['1-4094']}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}], 'private_attributes': [], 'short_name': None, 'state': 'enabled', 'status': 'up'}, {'id': 'urn:sdx:node:tenet.ac.za:Tenet03', 'location': {'address': 'EastLondon', 'iso3166_2_lvl4': 'ZA-EC', 'latitude': -33.0, 'longitude': 27.0}, 'name': 'Tenet03', 'ports': [{'entities': [], 'id': 'urn:sdx:port:tenet.ac.za:Tenet03:50', 'name': 'Tenet03-eth50', 'nni': '', 'node': 'urn:sdx:node:tenet.ac.za:Tenet03', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': ['1-4094']}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}, {'entities': [], 'id': 'urn:sdx:port:tenet.ac.za:Tenet03:2', 'name': 'Tenet03-eth2', 'nni': 'urn:sdx:link:tenet.ac.za:Tenet01/2_Tenet03/2', 'node': 'urn:sdx:node:tenet.ac.za:Tenet03', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': ['1-4094']}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}], 'private_attributes': [], 'short_name': None, 'state': 'enabled', 'status': 'up'}, {'id': 'urn:sdx:node:ampath.net:Ampath1', 'location': {'address': 'Miami', 'iso3166_2_lvl4': 'US-FL', 'latitude': 25.77, 'longitude': -80.19}, 'name': 'Ampath1', 'ports': [{'entities': [], 'id': 'urn:sdx:port:ampath.net:Ampath1:1', 'name': 'Ampath1-eth1', 'nni': 'urn:sdx:link:ampath.net:Ampath1/1_Ampath2/1', 'node': 'urn:sdx:node:ampath.net:Ampath1', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': [[1, 4094]]}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}, {'entities': [], 'id': 'urn:sdx:port:ampath.net:Ampath1:40', 'name': 'Ampath1-eth40', 'nni': 'urn:sdx:port:sax.net:Sax01:40', 'node': 'urn:sdx:node:ampath.net:Ampath1', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': [[1, 4094]]}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}, {'entities': [], 'id': 'urn:sdx:port:ampath.net:Ampath1:50', 'name': 'Ampath1-eth50', 'nni': '', 'node': 'urn:sdx:node:ampath.net:Ampath1', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': [[1, 4094]]}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}, {'entities': [], 'id': 'urn:sdx:port:ampath.net:Ampath1:2', 'name': 'Ampath1-eth2', 'nni': 'urn:sdx:link:ampath.net:Ampath1/2_Ampath3/2', 'node': 'urn:sdx:node:ampath.net:Ampath1', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': [[1, 4094]]}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}], 'private_attributes': [], 'short_name': None, 'state': 'enabled', 'status': 'up'}, {'id': 'urn:sdx:node:ampath.net:Ampath2', 'location': {'address': 'BocaRaton', 'iso3166_2_lvl4': 'US-FL', 'latitude': 26.38, 'longitude': -80.11}, 'name': 'Ampath2', 'ports': [{'entities': [], 'id': 'urn:sdx:port:ampath.net:Ampath2:1', 'name': 'Ampath2-eth1', 'nni': 'urn:sdx:link:ampath.net:Ampath1/1_Ampath2/1', 'node': 'urn:sdx:node:ampath.net:Ampath2', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': [[1, 4094]]}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}, {'entities': [], 'id': 'urn:sdx:port:ampath.net:Ampath2:50', 'name': 'Ampath2-eth50', 'nni': '', 'node': 'urn:sdx:node:ampath.net:Ampath2', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': [[1, 4094]]}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}, {'entities': [], 'id': 'urn:sdx:port:ampath.net:Ampath2:40', 'name': 'Ampath2-eth40', 'nni': 'urn:sdx:port:sax.net:Sax02:40', 'node': 'urn:sdx:node:ampath.net:Ampath2', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': [[1, 4094]]}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}, {'entities': [], 'id': 'urn:sdx:port:ampath.net:Ampath2:3', 'name': 'Ampath2-eth3', 'nni': 'urn:sdx:link:ampath.net:Ampath2/3_Ampath3/3', 'node': 'urn:sdx:node:ampath.net:Ampath2', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': [[1, 4094]]}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}], 'private_attributes': [], 'short_name': None, 'state': 'enabled', 'status': 'up'}, {'id': 'urn:sdx:node:ampath.net:Ampath3', 'location': {'address': 'Jacksonville', 'iso3166_2_lvl4': 'US-FL', 'latitude': 30.27, 'longitude': -81.68}, 'name': 'Ampath3', 'ports': [{'entities': [], 'id': 'urn:sdx:port:ampath.net:Ampath3:2', 'name': 'Ampath3-eth2', 'nni': 'urn:sdx:link:ampath.net:Ampath1/2_Ampath3/2', 'node': 'urn:sdx:node:ampath.net:Ampath3', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': [[1, 4094]]}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}, {'entities': [], 'id': 'urn:sdx:port:ampath.net:Ampath3:50', 'name': 'Ampath3-eth50', 'nni': '', 'node': 'urn:sdx:node:ampath.net:Ampath3', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': [[1, 4094]]}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}, {'entities': [], 'id': 'urn:sdx:port:ampath.net:Ampath3:3', 'name': 'Ampath3-eth3', 'nni': 'urn:sdx:link:ampath.net:Ampath2/3_Ampath3/3', 'node': 'urn:sdx:node:ampath.net:Ampath3', 'private_attributes': None, 'services': {'l2vpn_ptmp': {}, 'l2vpn_ptp': {'vlan_range': [[1, 4094]]}, 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'short_name': None, 'state': 'enabled', 'status': 'up', 'type': '10GE', 'vlan_range': None}], 'private_attributes': [], 'short_name': None, 'state': 'enabled', 'status': 'up'}], 'private_attributes': None, 'services': {'l2vpn_ptmp': [None], 'l2vpn_ptp': 'l2vpn-ptp', 'monitoring_capability': None, 'owner': None, 'private_attributes': None, 'provisioning_system': None, 'provisioning_url': None, 'vendor': None}, 'timestamp': '2026-04-08T17:52:56.997835', 'version': '2'} kytos_topo={'name': 'Ampath-OXP', 'id': 'urn:sdx:topology:ampath.net', 'version': 10, 'timestamp': '2026-04-08T17:53:35Z', 'model_version': '2.0.0', 'nodes': [{'name': 'Ampath1', 'id': 'urn:sdx:node:ampath.net:Ampath1', 'location': {'address': 'Miami', 'latitude': 25.77, 'longitude': -80.19, 'iso3166_2_lvl4': 'US-FL', 'private': []}, 'ports': [{'id': 'urn:sdx:port:ampath.net:Ampath1:1', 'name': 'Ampath1-eth1', 'node': 'urn:sdx:node:ampath.net:Ampath1', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:link:ampath.net:Ampath1/1_Ampath2/1', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath1:40', 'name': 'Ampath1-eth40', 'node': 'urn:sdx:node:ampath.net:Ampath1', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': '', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath1:50', 'name': 'Ampath1-eth50', 'node': 'urn:sdx:node:ampath.net:Ampath1', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': '', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath1:2', 'name': 'Ampath1-eth2', 'node': 'urn:sdx:node:ampath.net:Ampath1', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:link:ampath.net:Ampath1/2_Ampath3/2', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}], 'status': 'up', 'state': 'enabled'}, {'name': 'Ampath2', 'id': 'urn:sdx:node:ampath.net:Ampath2', 'location': {'address': 'BocaRaton', 'latitude': 26.38, 'longitude': -80.11, 'iso3166_2_lvl4': 'US-FL', 'private': []}, 'ports': [{'id': 'urn:sdx:port:ampath.net:Ampath2:1', 'name': 'Ampath2-eth1', 'node': 'urn:sdx:node:ampath.net:Ampath2', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:link:ampath.net:Ampath1/1_Ampath2/1', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath2:50', 'name': 'Ampath2-eth50', 'node': 'urn:sdx:node:ampath.net:Ampath2', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': '', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath2:40', 'name': 'Ampath2-eth40', 'node': 'urn:sdx:node:ampath.net:Ampath2', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:port:sax.net:Sax02:40', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath2:3', 'name': 'Ampath2-eth3', 'node': 'urn:sdx:node:ampath.net:Ampath2', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:link:ampath.net:Ampath2/3_Ampath3/3', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}], 'status': 'up', 'state': 'enabled'}, {'name': 'Ampath3', 'id': 'urn:sdx:node:ampath.net:Ampath3', 'location': {'address': 'Jacksonville', 'latitude': 30.27, 'longitude': -81.68, 'iso3166_2_lvl4': 'US-FL', 'private': []}, 'ports': [{'id': 'urn:sdx:port:ampath.net:Ampath3:2', 'name': 'Ampath3-eth2', 'node': 'urn:sdx:node:ampath.net:Ampath3', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:link:ampath.net:Ampath1/2_Ampath3/2', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath3:50', 'name': 'Ampath3-eth50', 'node': 'urn:sdx:node:ampath.net:Ampath3', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': '', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath3:3', 'name': 'Ampath3-eth3', 'node': 'urn:sdx:node:ampath.net:Ampath3', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:link:ampath.net:Ampath2/3_Ampath3/3', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}], 'status': 'up', 'state': 'enabled'}], 'links': [{'name': 'Ampath1/1_Ampath2/1', 'id': 'urn:sdx:link:ampath.net:Ampath1/1_Ampath2/1', 'ports': ['urn:sdx:port:ampath.net:Ampath1:1', 'urn:sdx:port:ampath.net:Ampath2:1'], 'type': 'intra', 'bandwidth': 10, 'residual_bandwidth': 100, 'latency': 0, 'packet_loss': 0, 'availability': 100, 'status': 'up', 'state': 'enabled', 'private': ['packet_loss']}, {'name': 'Ampath2/3_Ampath3/3', 'id': 'urn:sdx:link:ampath.net:Ampath2/3_Ampath3/3', 'ports': ['urn:sdx:port:ampath.net:Ampath2:3', 'urn:sdx:port:ampath.net:Ampath3:3'], 'type': 'intra', 'bandwidth': 10, 'residual_bandwidth': 100, 'latency': 0, 'packet_loss': 0, 'availability': 100, 'status': 'up', 'state': 'enabled', 'private': ['packet_loss']}, {'name': 'Ampath1/2_Ampath3/2', 'id': 'urn:sdx:link:ampath.net:Ampath1/2_Ampath3/2', 'ports': ['urn:sdx:port:ampath.net:Ampath1:2', 'urn:sdx:port:ampath.net:Ampath3:2'], 'type': 'intra', 'bandwidth': 10, 'residual_bandwidth': 100, 'latency': 0, 'packet_loss': 0, 'availability': 100, 'status': 'up', 'state': 'enabled', 'private': ['packet_loss']}], 'services': ['l2vpn-ptp']}
E                       assert 'urn:sdx:port....net:Sax01:40' == ''
E                         + urn:sdx:port:sax.net:Sax01:40

tests/test_21_use_case_topology.py:356: AssertionError
-------------------------------------------------------------------------------------------------- Captured stderr setup ---------------------------------------------------------------------------------------------------
*** Error setting resource limits. Mininet's performance may be affected.
---------------------------------------------------------------------------------------------------- Captured log setup ----------------------------------------------------------------------------------------------------
WARNING  mininet:log.py:153 *** Error setting resource limits. Mininet's performance may be affected.
----------------------------------------------------------------------------------------------------- start/stop times -----------------------------------------------------------------------------------------------------
tests/test_21_use_case_topology.py::TestE2ETopologyUseCases::test_091_port_missing_nni: 2026-04-08,17:53:27.563221 - 2026-04-08,17:53:41.892444
================================================================================================= short test summary info ==================================================================================================
FAILED tests/test_21_use_case_topology.py::TestE2ETopologyUseCases::test_091_port_missing_nni - AssertionError: sdx_topo={'id': 'urn:sdx:topology:', 'links': [{'availability': 100, 'bandwidth': 10, 'id': 'urn:sdx:link:interdomain:sax.net:Sax01:41:tenet.ac.za:Tenet01:41', 'latency': 0, 'measurement_period': Non...
=============================================================================================== 1 failed in 73.34s (0:01:13) ===============================================================================================
```

Then if you check Kytos's logs:

```
$ docker compose cp ampath:/var/log/syslog ampath.log
$ grep -i troubleshooting ampath.log
2026-04-08T17:53:04.373370+00:00 9b7e8b2892d7 kytos.napps.kytos/sdx:INFO main:417:  troubleshooting: topology={'name': 'Ampath-OXP', 'id': 'urn:sdx:topology:ampath.net', 'version': 9, 'timestamp': '2026-04-08T17:52:56Z', 'model_version': '2.0.0', 'nodes': [{'name': 'Ampath1', 'id': 'urn:sdx:node:ampath.net:Ampath1', 'location': {'address': 'Miami', 'latitude': 25.77, 'longitude': -80.19, 'iso3166_2_lvl4': 'US-FL', 'private': []}, 'ports': [{'id': 'urn:sdx:port:ampath.net:Ampath1:1', 'name': 'Ampath1-eth1', 'node': 'urn:sdx:node:ampath.net:Ampath1', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:link:ampath.net:Ampath1/1_Ampath2/1', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath1:40', 'name': 'Ampath1-eth40', 'node': 'urn:sdx:node:ampath.net:Ampath1', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:port:sax.net:Sax01:40', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath1:50', 'name': 'Ampath1-eth50', 'node': 'urn:sdx:node:ampath.net:Ampath1', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': '', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath1:2', 'name': 'Ampath1-eth2', 'node': 'urn:sdx:node:ampath.net:Ampath1', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:link:ampath.net:Ampath1/2_Ampath3/2', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}], 'status': 'up', 'state': 'enabled'}, {'name': 'Ampath2', 'id': 'urn:sdx:node:ampath.net:Ampath2', 'location': {'address': 'BocaRaton', 'latitude': 26.38, 'longitude': -80.11, 'iso3166_2_lvl4': 'US-FL', 'private': []}, 'ports': [{'id': 'urn:sdx:port:ampath.net:Ampath2:1', 'name': 'Ampath2-eth1', 'node': 'urn:sdx:node:ampath.net:Ampath2', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:link:ampath.net:Ampath1/1_Ampath2/1', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath2:50', 'name': 'Ampath2-eth50', 'node': 'urn:sdx:node:ampath.net:Ampath2', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': '', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath2:40', 'name': 'Ampath2-eth40', 'node': 'urn:sdx:node:ampath.net:Ampath2', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:port:sax.net:Sax02:40', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath2:3', 'name': 'Ampath2-eth3', 'node': 'urn:sdx:node:ampath.net:Ampath2', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:link:ampath.net:Ampath2/3_Ampath3/3', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}], 'status': 'up', 'state': 'enabled'}, {'name': 'Ampath3', 'id': 'urn:sdx:node:ampath.net:Ampath3', 'location': {'address': 'Jacksonville', 'latitude': 30.27, 'longitude': -81.68, 'iso3166_2_lvl4': 'US-FL', 'private': []}, 'ports': [{'id': 'urn:sdx:port:ampath.net:Ampath3:2', 'name': 'Ampath3-eth2', 'node': 'urn:sdx:node:ampath.net:Ampath3', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:link:ampath.net:Ampath1/2_Ampath3/2', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath3:50', 'name': 'Ampath3-eth50', 'node': 'urn:sdx:node:ampath.net:Ampath3', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': '', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath3:3', 'name': 'Ampath3-eth3', 'node': 'urn:sdx:node:ampath.net:Ampath3', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:link:ampath.net:Ampath2/3_Ampath3/3', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}], 'status': 'up', 'state': 'enabled'}], 'links': [{'name': 'Ampath1/1_Ampath2/1', 'id': 'urn:sdx:link:ampath.net:Ampath1/1_Ampath2/1', 'ports': ['urn:sdx:port:ampath.net:Ampath1:1', 'urn:sdx:port:ampath.net:Ampath2:1'], 'type': 'intra', 'bandwidth': 10, 'residual_bandwidth': 100, 'latency': 0, 'packet_loss': 0, 'availability': 100, 'status': 'up', 'state': 'enabled', 'private': ['packet_loss']}, {'name': 'Ampath2/3_Ampath3/3', 'id': 'urn:sdx:link:ampath.net:Ampath2/3_Ampath3/3', 'ports': ['urn:sdx:port:ampath.net:Ampath2:3', 'urn:sdx:port:ampath.net:Ampath3:3'], 'type': 'intra', 'bandwidth': 10, 'residual_bandwidth': 100, 'latency': 0, 'packet_loss': 0, 'availability': 100, 'status': 'up', 'state': 'enabled', 'private': ['packet_loss']}, {'name': 'Ampath1/2_Ampath3/2', 'id': 'urn:sdx:link:ampath.net:Ampath1/2_Ampath3/2', 'ports': ['urn:sdx:port:ampath.net:Ampath1:2', 'urn:sdx:port:ampath.net:Ampath3:2'], 'type': 'intra', 'bandwidth': 10, 'residual_bandwidth': 100, 'latency': 0, 'packet_loss': 0, 'availability': 100, 'status': 'up', 'state': 'enabled', 'private': ['packet_loss']}], 'services': ['l2vpn-ptp']}
2026-04-08T17:53:35.824254+00:00 9b7e8b2892d7 kytos.napps.kytos/sdx:INFO main:417:  troubleshooting: topology={'name': 'Ampath-OXP', 'id': 'urn:sdx:topology:ampath.net', 'version': 9, 'timestamp': '2026-04-08T17:52:56Z', 'model_version': '2.0.0', 'nodes': [{'name': 'Ampath1', 'id': 'urn:sdx:node:ampath.net:Ampath1', 'location': {'address': 'Miami', 'latitude': 25.77, 'longitude': -80.19, 'iso3166_2_lvl4': 'US-FL', 'private': []}, 'ports': [{'id': 'urn:sdx:port:ampath.net:Ampath1:1', 'name': 'Ampath1-eth1', 'node': 'urn:sdx:node:ampath.net:Ampath1', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:link:ampath.net:Ampath1/1_Ampath2/1', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath1:40', 'name': 'Ampath1-eth40', 'node': 'urn:sdx:node:ampath.net:Ampath1', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:port:sax.net:Sax01:40', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath1:50', 'name': 'Ampath1-eth50', 'node': 'urn:sdx:node:ampath.net:Ampath1', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': '', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath1:2', 'name': 'Ampath1-eth2', 'node': 'urn:sdx:node:ampath.net:Ampath1', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:link:ampath.net:Ampath1/2_Ampath3/2', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}], 'status': 'up', 'state': 'enabled'}, {'name': 'Ampath2', 'id': 'urn:sdx:node:ampath.net:Ampath2', 'location': {'address': 'BocaRaton', 'latitude': 26.38, 'longitude': -80.11, 'iso3166_2_lvl4': 'US-FL', 'private': []}, 'ports': [{'id': 'urn:sdx:port:ampath.net:Ampath2:1', 'name': 'Ampath2-eth1', 'node': 'urn:sdx:node:ampath.net:Ampath2', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:link:ampath.net:Ampath1/1_Ampath2/1', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath2:50', 'name': 'Ampath2-eth50', 'node': 'urn:sdx:node:ampath.net:Ampath2', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': '', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath2:40', 'name': 'Ampath2-eth40', 'node': 'urn:sdx:node:ampath.net:Ampath2', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:port:sax.net:Sax02:40', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath2:3', 'name': 'Ampath2-eth3', 'node': 'urn:sdx:node:ampath.net:Ampath2', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:link:ampath.net:Ampath2/3_Ampath3/3', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}], 'status': 'up', 'state': 'enabled'}, {'name': 'Ampath3', 'id': 'urn:sdx:node:ampath.net:Ampath3', 'location': {'address': 'Jacksonville', 'latitude': 30.27, 'longitude': -81.68, 'iso3166_2_lvl4': 'US-FL', 'private': []}, 'ports': [{'id': 'urn:sdx:port:ampath.net:Ampath3:2', 'name': 'Ampath3-eth2', 'node': 'urn:sdx:node:ampath.net:Ampath3', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:link:ampath.net:Ampath1/2_Ampath3/2', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath3:50', 'name': 'Ampath3-eth50', 'node': 'urn:sdx:node:ampath.net:Ampath3', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': '', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}, {'id': 'urn:sdx:port:ampath.net:Ampath3:3', 'name': 'Ampath3-eth3', 'node': 'urn:sdx:node:ampath.net:Ampath3', 'type': '10GE', 'status': 'up', 'state': 'enabled', 'mtu': 1500, 'nni': 'urn:sdx:link:ampath.net:Ampath2/3_Ampath3/3', 'services': {'l2vpn-ptp': {'vlan_range': [[1, 4094]]}}, 'private': ['status'], 'entities': []}], 'status': 'up', 'state': 'enabled'}], 'links': [{'name': 'Ampath1/1_Ampath2/1', 'id': 'urn:sdx:link:ampath.net:Ampath1/1_Ampath2/1', 'ports': ['urn:sdx:port:ampath.net:Ampath1:1', 'urn:sdx:port:ampath.net:Ampath2:1'], 'type': 'intra', 'bandwidth': 10, 'residual_bandwidth': 100, 'latency': 0, 'packet_loss': 0, 'availability': 100, 'status': 'up', 'state': 'enabled', 'private': ['packet_loss']}, {'name': 'Ampath2/3_Ampath3/3', 'id': 'urn:sdx:link:ampath.net:Ampath2/3_Ampath3/3', 'ports': ['urn:sdx:port:ampath.net:Ampath2:3', 'urn:sdx:port:ampath.net:Ampath3:3'], 'type': 'intra', 'bandwidth': 10, 'residual_bandwidth': 100, 'latency': 0, 'packet_loss': 0, 'availability': 100, 'status': 'up', 'state': 'enabled', 'private': ['packet_loss']}, {'name': 'Ampath1/2_Ampath3/2', 'id': 'urn:sdx:link:ampath.net:Ampath1/2_Ampath3/2', 'ports': ['urn:sdx:port:ampath.net:Ampath1:2', 'urn:sdx:port:ampath.net:Ampath3:2'], 'type': 'intra', 'bandwidth': 10, 'residual_bandwidth': 100, 'latency': 0, 'packet_loss': 0, 'availability': 100, 'status': 'up', 'state': 'enabled', 'private': ['packet_loss']}], 'services': ['l2vpn-ptp']}
```

To summarize the logs above: when we run a `GET /api/kytos/sdx/topology/2.0.0` from Kytos Ampath OXP, we get version 10, but when we ran `POST /api/kytos/sdx/topology/2.0.0`, the version that we sent to SDX-LC was version 9.

--- a/main.py
+++ b/main.py
@@ -407,14 +407,16 @@ class Main(KytosNApp):  # pylint: disable=R0904
     @rest("topology/2.0.0", methods=["GET"])
     def get_sdx_topology_v2(self, _request: Request) -> JSONResponse:
         """return sdx topology v2"""
-        if not self._converted_topo.get("nodes"):
-            return JSONResponse({}, status_code=200)
-        return JSONResponse(self._converted_topo)
+        with self._topo_lock:
+            if not self._converted_topo.get("nodes"):
+                return JSONResponse({}, status_code=200)
+            return JSONResponse(self._converted_topo)
 
     @rest("topology/2.0.0", methods=["POST"])
     def send_topology_to_sdxlc(self, _request: Request) -> JSONResponse:
         """Send the topology (v2) to SDX-LC"""
-        self.post_topology_to_sdxlc(self._converted_topo)
+        with self._topo_lock:
+            self.post_topology_to_sdxlc(self._converted_topo)
         return JSONResponse("Operation successful", status_code=200)
 
     @rest("l2vpn/1.0", methods=["POST"])

--- a/settings.py
+++ b/settings.py
@@ -14,7 +14,7 @@ OXPO_URL = "testoxp.net"
 
 # TOPOLOGY_EVENT_WAIT: time to wait while hanlde topology update
 # events to try to group them
-TOPOLOGY_EVENT_WAIT = 5
+TOPOLOGY_EVENT_WAIT = 3
 
 # Kytos mef_eline endpoint for creating L2VPN PTP
 KYTOS_EVC_URL = "http://127.0.0.1:8181/api/kytos/mef_eline/v2/evc/"

